src/rgw/rgw_user.cc

```diff
@@ -636,8 +636,10 @@ int RGWAccessKeyPool::check_op(RGWUserAdminOpState& op_state,
 
   // don't check for secret key because we may be doing a removal
 
-  check_existing_key(op_state);
-
+  if (check_existing_key(op_state)) {
+    set_err_msg(err_msg, "cannot create existing key");
+    return -ERR_KEY_EXIST;
+  }
```

Comment:
this is causing a test failure here: https://github.com/ceph/ceph/blob/master/qa/tasks/radosgw_admin.py#L404-L408

```bash
$ radosgw-admin key rm --uid foo --access-key p5YnriCv1nAtykxBrupQ
could not remove key: unable to parse request, cannot create existing key
```
you can find the full logs here: http://qa-proxy.ceph.com/teuthology/cbodley-2021-07-19_19:01:44-rgw-wip-cbodley-testing-distro-basic-smithi/6279602/teuthology.log